traffic_ops/testing/api_contract/v4/conftest.py

```diff
@@ -628,3 +628,187 @@ def profile_post_data(to_session: TOSession, request_template_data: list[JSONDat
+		logger.error("No division response data from division POST request.")
+		sys.exit(1)
+
+@pytest.fixture()
+def region_post_data(to_session: TOSession, request_template_data: list[JSONData]
+		  ) -> dict[str, object]:
+	"""
+	PyTest Fixture to create POST data for regions endpoint.
+
+	:param to_session: Fixture to get Traffic Ops session.
+	:param request_template_data: Fixture to get regions request template data from
+	request_template file.
+	:returns: Sample POST data and the actual API response.
+	"""
+
+	try:
+		region = request_template_data[0]
+	except IndexError as e:
+		raise TypeError(
+			"malformed prerequisite data; no region present in 'regions' array property") from e
+
+	if not isinstance(region, dict):
+		raise TypeError(
+			f"malformed prerequisite data; regions must be objects, not '{type(region)}'")
+
+	# Return new post data and post response from region POST request
+	randstr = str(randint(0, 1000))
+	try:
+		name = region["name"]
+		if not isinstance(name, str):
+			raise TypeError(f"name must be str, not '{type(name)}'")
+		region["name"] = name[:4] + randstr
+		division = region["division"]
+		if not isinstance(division, str):
+			raise TypeError(f"division must be int, not '{type(division)}'")
+		region["division"] = division[:4] + randstr
+		divisionname = region["divisionname"]
+		if not isinstance(divisionname, str):
+			raise TypeError(f"divisionname must be str, not '{type(divisionname)}'")
+		region["divisionname"] = divisionname[:4] + randstr
+	except KeyError as e:
+		raise TypeError(f"missing Parameter property '{e.args[0]}'") from e
+
+	logger.info("New region data to hit POST method %s", request_template_data)
+	# Hitting region POST methed
+	response: tuple[JSONData, requests.Response] = to_session.create_region(data=region)
+	try:
+		resp_obj = response[0]
+		if not isinstance(resp_obj, dict):
+			raise TypeError("malformed API response; region is not an object")
+		return resp_obj
+	except IndexError:
+		logger.error("No region response data from region POST request.")
+		sys.exit(1)
+
+@pytest.fixture()
+def phys_locations__post_data(to_session: TOSession, request_template_data: list[JSONData]
+		  ) -> dict[str, object]:
+	"""
+	PyTest Fixture to create POST data for phys_locations endpoint.
+
+	:param to_session: Fixture to get Traffic Ops session.
+	:param request_template_data: Fixture to get phys_locations request template data from
+	request_template file.
+	:returns: Sample POST data and the actual API response.
+	"""
+
+	try:
+		phys_location = request_template_data[0]
+	except IndexError as e:
+		raise TypeError(
+			"malformed prerequisite data; no phys_location in 'phys_locations' array property") from e
+
+	if not isinstance(phys_location, dict):
+		raise TypeError(
+			f"malformed prerequisite data; phys_locations must be objects, not '{type(phys_location)}'")
+
+	# Return new post data and post response from phys_locations POST request
+	randstr = str(randint(0, 1000))
+	try:
```

Comment:
why you generating new data instead of using post api

Reply:
Changed to use post api.